code/game/objects/items/devices/scanners/sequence_scanner.dm

```diff
@@ -31,6 +38,31 @@
 	else
 		user.visible_message(span_notice("[user] fails to analyze [target]'s genetic sequence."), span_warning("[target] has no readable genetic sequence!"))
 
+/obj/item/sequence_scanner/attack_secondary(mob/living/target, mob/living/carbon/human/user, max_interact_count=1)
+	add_fingerprint(user)
+	//no scanning if its a husk or DNA-less Species
+	if (!HAS_TRAIT(target, TRAIT_GENELESS) && !HAS_TRAIT(target, TRAIT_BADDNA))
+		user.visible_message(span_warning("[user] is scanning [target]'s genetic makeup."))
+		if(!do_after(user, 3 SECONDS))
+			balloon_alert(user, "scan failed!")
+			user.visible_message(span_warning("[user] fails to scan [target]'s genetic makeup."))
+			return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+		makeup_scan(target, user)
+		balloon_alert(user, "makeup scanned")
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+	else
+		user.visible_message(span_notice("[user] fails to analyze [target]'s genetic makeup."), span_warning("[target] has no readable genetic makeup!"))
+		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/item/sequence_scanner/afterattack_secondary(obj/object, mob/user, proximity)
+	. = ..()
+	if(!istype(object) || !proximity)
+		return
+	if(istype(object, /obj/machinery/computer/scan_consolenew))
+		var/obj/machinery/computer/scan_consolenew/console = object
+		var/buffer_index = tgui_input_number(user, "Slot:", "Which slot to export:", 1, LAZYLEN(console.genetic_makeup_buffer), 1)
```

Comment:
Never trust player input. Make sure it's a valid input first (so we dont get people adding 7000 makeup buffers)

Reply:
It's already ranged by the amount of slots in the dna console buffer

Reply:
Oh right, this isn't the byond thing